### ISSUE TITLE
Does not vibrate at end of stopwatch or alarm

### ISSUE THREAD

**Issue by dos286** [dakhnod/Fossil-HR-Timer#1]
The watch does not vibrate at the end of the better stopwatch, or better alarm.

**Comment by dakhnod** [comment]
Interesting. There should be a notification played. Do you have your watch on DND? Do regular notifications work?

**Comment by dos286** [comment]
I have the watch DND setting on "mirror phone" and the phone is not on DND. Typically, alarms and timers will vibrate even with DND on.
I tried setting DND to "off" and there was still no vibration.
Does the notification run through the companion app? I did not get a notification on my phone either.

**Comment by dakhnod** [comment]
The timer app does not vibrate the watch by itself. It plays a notification on the watch when the timer finishes. Hence my suspicion with DND.
In any case, what watch are you using?

**Comment by dos286** [comment]
Fossil Hybrid HR Collider.

**Comment by dakhnod** [comment]
So, do regular notifications work on your watch right now?

**Comment by dos286** [comment]
Yes. I'm getting regular notifications. I have a limited set of apps whitelisted in gadgetbridge that pass notifications to the watch. Do I need to whitelist the companion app?

**Comment by arjan-s** [comment]
I experienced the same since updating to the major new DN1.0.3.* firmware release

**Comment by dos286** [comment]
That makes sense to me.
I'm a first time Gadgetbridge user since (because of) the major new update, so I'm not certain if it would have worked on my watch with the previous firmware.

**Comment by dakhnod** [comment]
So it's an issue with the firmware version, too bad. I am not planning on updating to the newest firmware...

@arjan-s could you figure out from the newer apps how to send notifications?

**Comment by arjan-s** [comment]
Yes, I'll take a look

**Comment by arjan-s** [comment]
I've looked at the code of the new timer app, and I'm afraid it sends the urgent_notification the same way as the old app. So I have no clue yet why it doesn't work.

**Comment by dakhnod** [comment]
there is a new branch that should fix vibrations.

We lose the notification ations though, no idea how to do them since I couldn't find any new apps posting notifications.

@arjan-s could you please check again, since my research revealed that the call is actually urgend_notification_new?

**Comment by arjan-s** [comment]
I've sent you my decompiled JS via mail, can you take a look?
I was wrong about the `urgent_notification` in the new app, I was actually looking at the code of the old app.
The new app doesn't use an urgent notification anymore at all. It now uses `response.vibe = "timer"` instead, repeating every 800ms, and as you said there's no actions anymore except going back to home.

Edit: email didn't work, so here's the decompiled app: https://pastebin.com/uyFCG4wZ

**Comment by dakhnod** [comment]
I disagree. For me urgent_notification_new works...

**Comment by arjan-s** [comment]
I know it works :)
I'm just saying that the latest version of the timerApp doesn't use it. But the new stopwatchApp and wellnessApp do use it. I didn't decompile those, so I don't know for sure how they use it and whether there are still actions possible.

**Comment by gjkrediet** [comment]
A practical solution could be to bring the app to the foreground for notification/vibration which worked OK for me. If helpfull I could share the changes in the code to accomplish that.

**Comment by dakhnod** [comment]
@gjkrediet does it not work for you?

**Comment by gjkrediet** [comment]
The timer I found n Your SDK (lots of thanks) does only work for me in older versions of the firmware. I tried urgent_notification_new which only brought up a notification for a short time but no vibration. I found in the latest navigationApp by @arjan-s how to bring the timer to te foreground and actualy that works better for me. I made the same kind of timer for a pebble with the same approach and found it to be a good solution. Does that answer your question?

**Comment by dakhnod** [comment]
@gjkrediet you made the pebble timer? I probably stole your idea then, plus the alarm function :)

**Comment by gjkrediet** [comment]
You are referring to the pebble timer+ which has a  stopwatch. I am not to credit for that (despite my name). I think I took a copy of it and adjusted it to my needs. Also meaning I deleted the stopwatch-functionality which I found too complex in using. But this is all years ago. Later I made my own timer for Garmin vivoactive. The factory-timer is hardly useable. I did the same trick: around the time it should alarm it comes to the foreground which gives more control, like the possibility to show how many seconds have passed since the alarm went off. Same now with the fossil.
